### PR TITLE
Design clean-up and consistency across expense reports and requests

### PR DESCRIPTION
**Pull request by mountiny**
### Details

The app has several inaccuracies for previewing a description and merchant of expense reports and expense requests. Also, we are fixing the whisper effect on messages and removing right carets form all report and request previews. 
**NOTE:** the initial proposal has more items to implement, but not all of them were decided to fix. Please read the **"Tests"** block below to check all fixed scenarios. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/26138
PROPOSAL: https://github.com/Expensify/App/issues/26138#issuecomment-1701226606

#### 1. We're showing right **carets** for all **report** and **request previews**, when instead right **carets** should be removed from all **request** and report **previews**.
1. Open any workspace chat.
2. Request any type/amount of money requests.
3. Verify that there is no any right carets.

#### 2. Similarly, `IOU` **reports** are showing the **merchant** when they should instead be showing the **description** (_because `IOUs` always have the same **merchant**, which is **Request_**).
1. Open any workspace chat.
2. Request default cash requests with/without description.
3. Verify that there is no merchant on requests.
4. Verify there is a description on requests where you've added it.

#### 3. When a **merchant** name exists for an **expense** / **request preview**, we only show the **merchant**.
1. Open any workspace chat.
2. Request a scan receipt requests with/without description.
3. Verify that there is no description on requests.
4. Verify there is a merchant on requests.

#### 4. Normal "Cash" requests are missing the `X requests` logic.
1. Open any workspace chat.
2. Request a few times default cash requests.
3. Verify that starts from 2+ requests, there is a counter.

#### 5. Receipt whisper cards are no visible when they are inside "Only visible for you" message.
1. Open any workspace chat.
2. Request a receipt request.
3. Verify that their cards are contrast with the background.

#### 6. (none) is showing as a merchant for receipts being scanned
1. Open any workspace chat.
2. Request a receipt request without customizing the merchant
3. Verify that the card **does not** show `(none)` as a merchant
4. Click into the detail of the request
5. Verify the Merchant is `(none)`

#### 7. Rounded bottom corners of the images in the report preview
1. Open any workspace chat.
2. Make a distance or receipt request
3. Verify that the report preview and the request preview inside the expense report has bottom corners rounded as well

#### 8. On Report Preview for requests made to Saastr or SBE workspaces, there is no user email in the workspace name
1. Open workspace chat with SaaStr or SBE workspace
4. Make a request to such workspace
5. Verify that the message says `SBE owes $xx.xx` or `SaaStr owes $xx.xx` and there is no email prepending the SBE

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as **"Tests"**

### QA Steps

Same as **"Tests"**

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

#### 1. We're showing right **carets** for all **report** and **request previews**, when instead right **carets** should be removed from all **request** and report **previews**.
1 Safari
1 1 Safari

1 Chrome
1 1 Chrome

#### 2. Similarly, `IOU` **reports** are showing the **merchant** when they should instead be showing the **description** (_because `IOUs` always have the same **merchant**, which is **Request_**).
2 Safari

2 Chrome

#### 3. When a **merchant** name exists for an **expense** / **request preview**, we only show the **merchant**.
3 Safari

3 Chrome

#### 4. Normal "Cash" requests are missing the `X requests` logic.
4 Safari

4 Chrome

#### 5. Cards are no visible when they are inside "Only visible for you" message.
5 Safari
5 1 Safari

5 Chrome
5 1 Chrome

#### 6. (none) is not showing for the smartscanning receipts

image

#### 7. Rounded bottom corners of the images in the report preview

image

#### 8. On Report Preview for requests made to Saastr or SBE workspaces, there is no user email in the workspace name

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

#### 1. We're showing right **carets** for all **report** and **request previews**, when instead right **carets** should be removed from all **request** and report **previews**.
1 IOS Chrome
1 1 IOS Chrome

#### 2. Similarly, `IOU` **reports** are showing the **merchant** when they should instead be showing the **description** (_because `IOUs` always have the same **merchant**, which is **Request_**).
2 IOS Chrome

#### 3. When a **merchant** name exists for an **expense** / **request preview**, we only show the **merchant**.
3 IOS Chrome

#### 4. Normal "Cash" requests are missing the `X requests` logic.
4 IOS Chrome

#### 5. Cards are no visible when they are inside "Only visible for you" message.
5 IOS Chrome
5 1 IOS Chrome

</details>

<details>
<summary>Mobile Web - Safari</summary>

#### 1. We're showing right **carets** for all **report** and **request previews**, when instead right **carets** should be removed from all **request** and report **previews**.
1 IOS Safari

1 1 IOS Safari

#### 2. Similarly, `IOU` **reports** are showing the **merchant** when they should instead be showing the **description** (_because `IOUs` always have the same **merchant**, which is **Request_**).
2 IOS Safari

#### 3. When a **merchant** name exists for an **expense** / **request preview**, we only show the **merchant**.
3 IOS Safari

#### 4. Normal "Cash" requests are missing the `X requests` logic.
4 IOS Safari

#### 5. Cards are no visible when they are inside "Only visible for you" message.
5 IOS Safari

5 1 IOS Safari

</details>

<details>
<summary>Desktop</summary>

#### 1. We're showing right **carets** for all **report** and **request previews**, when instead right **carets** should be removed from all **request** and report **previews**.
1 Desktop
1 1 Desktop

#### 2. Similarly, `IOU` **reports** are showing the **merchant** when they should instead be showing the **description** (_because `IOUs` always have the same **merchant**, which is **Request_**).
2 Desktop

#### 3. When a **merchant** name exists for an **expense** / **request preview**, we only show the **merchant**.
3 Desktop

#### 4. Normal "Cash" requests are missing the `X requests` logic.
4 Desktop

#### 5. Cards are no visible when they are inside "Only visible for you" message.
5 Desktop
5 1 Desktop

</details>

<details>
<summary>iOS</summary>

#### 1. We're showing right **carets** for all **report** and **request previews**, when instead right **carets** should be removed from all **request** and report **previews**.
1 IOS
1 1 IOS

#### 2. Similarly, `IOU` **reports** are showing the **merchant** when they should instead be showing the **description** (_because `IOUs` always have the same **merchant**, which is **Request_**).
2 IOS

#### 3. When a **merchant** name exists for an **expense** / **request preview**, we only show the **merchant**.
3 IOS

#### 4. Normal "Cash" requests are missing the `X requests` logic.
4 IOS

#### 5. Cards are no visible when they are inside "Only visible for you" message.
5 IOS
5 1 IOS

</details>

<details>
<summary>Android</summary>

#### 1. We're showing right **carets** for all **report** and **request previews**, when instead right **carets** should be removed from all **request** and report **previews**.
1 Android
1 1 Android

#### 2. Similarly, `IOU` **reports** are showing the **merchant** when they should instead be showing the **description** (_because `IOUs` always have the same **merchant**, which is **Request_**).
2 Android

#### 3. When a **merchant** name exists for an **expense** / **request preview**, we only show the **merchant**.
3 Android

#### 4. Normal "Cash" requests are missing the `X requests` logic.
4 Android

#### 5. Cards are no visible when they are inside "Only visible for you" message.
5 Android
5 1 Android

</details>

### CODE REVIEW

**Comment by situchan** [comment]
I see @rezkiy37 raised [PR](https://github.com/Expensify/App/pull/26503) already. Is it different purpose?

**Comment by JmillsExpensify** [comment]
@mountiny for the requests. They are missing the you in "Only visible to you"

Also did we decide to not update the receipt count overly you the bottom right for now? That's fine given the urgency of this PR, just wanted to confirm it's on purpose.

**Comment by JmillsExpensify** [comment]
This is missing a "you" 

image

**Comment by JmillsExpensify** [comment]
Then this case. image

This is the `expense` case and not the `IOU` case, just for clarity. So as a result, there is no default `request` merchant right?

**Comment by situchan** [comment]
Optimistic data seems broken on main
bug

**Comment by JmillsExpensify** [comment]
For the receipt overlay, when they are more than three receipts, it should be [this style](https://github.com/Expensify/App/issues/26138#issuecomment-1700918711)

**Comment by mountiny** [comment]
> Also did we decide to not update the receipt count overly you the bottom right for now? That's fine given the urgency of this PR, just wanted to confirm it's on purpose.

Yep I think that was omitted here and I agree what we have now is fine 

> For the receipt overlay, when there are more than three receipts, it should be https://github.com/Expensify/App/issues/26138#issuecomment-1700918711.

as mentioned above, I think this is a polish we can leave for later.

Expense cash request preview with default `Request` merchant:
image

IOU cash request preview with default `Request` merchant Not showing and no description being added either:
image


Only visible to you, I am getting that correctly, not sure how Michael got that :D 
image

**Comment by mountiny** [comment]
@situchan Add the tests and screenshots for:
- email not being part of the saastr or sbe email in the report preview
- the receipt or distance request previews are rounded at bottom as well

**Comment by JmillsExpensify** [comment]
Cool, looks good. @mountiny maybe we keep the IOU and expense logic the same so we don't create inconsistencies? Eg

- `Request` auto merchant don't show 
- if description show
- if neither of those apply show merchant

**Comment by situchan** [comment]
SAASTR test result:

Screenshot 2023-09-03 at 9 11 20 PM

Screenshot 2023-09-03 at 9 11 26 PM


Screenshot 2023-09-03 at 9 11 11 PM

**Comment by situchan** [comment]
`none` doesn't need localization?

Screenshot 2023-09-03 at 9 25 38 PM

**Comment by mountiny** [comment]
> none doesn't need localization?


Correct thats a value in database, no need to translate

> Cool, looks good. @mountiny maybe we keep the IOU and expense logic the same so we don't create inconsistencies? 

Updated so it does not matter if its expense or iou, it shows merchant if its custom (not `Request` and not `(none)`), if thats not true, we should description if set

**Comment by JmillsExpensify** [comment]
Perfect thanks!

File: src/libs/ReportUtils.js [hunk 1]
```diff
@@ -124,7 +124,15 @@ function getPolicyName(report, returnEmptyIfNotFound = false, policy = undefined
 
     // Public rooms send back the policy name with the reportSummary,
     // since they can also be accessed by people who aren't in the workspace
-    return lodashGet(finalPolicy, 'name') || report.policyName || report.oldPolicyName || noPolicyFound;
+    const policyName = lodashGet(finalPolicy, 'name') || report.policyName || report.oldPolicyName || noPolicyFound;
+
+    // The SBE and SAASTR policies have the user name in its name, however, we do not want to show that
+    if (lodashGet(finalPolicy, 'owner') === CONST.EMAIL.SBE || lodashGet(finalPolicy, 'owner') === CONST.EMAIL.SAASTR) {
+        const policyNameParts = policyName.split(' ');
+        if (!Str.isValidEmail(policyNameParts[0])) return policyName;
```
Comment (situchan) [inline src/libs/ReportUtils.js:132]: How about phone number accounts?

**Review by situchan — COMMENTED** [review]
Lint failing

**Comment by situchan** [comment]
Not from this PR, but I can do endless payment with $0. After click Pay elsewhere, it says paid and then Pay button reappears. This is repeated


https://github.com/Expensify/App/assets/108292595/74e39a61-49aa-4a66-849e-95476ff1f17c

File: src/libs/ReportUtils.js (hunk 1 above)
Comment (mountiny) [reply src/libs/ReportUtils.js:132]: Tested now and seems to be working fine for me locally!

Comment (mountiny) [reply src/libs/ReportUtils.js:132]: its in the @expensify.sms format

**Comment by mountiny** [comment]
> Not from this PR, but I can do endless payment with $0. After click Pay elsewhere, it says paid and then Pay button reappears. This is repeated

So this is an admin on its own workspace right? 

I think this will be because the smartscans are not valid and we dont handle the errors well. Its not critical for the conference

**Comment by mountiny** [comment]
Pushed in some small clean up I missed before

**Comment by situchan** [comment]
This is super minor but it looks nicer when image border color is same as bottom background when hovered (or whisper) so that it feels image is fully inside card view

Screenshot 2023-09-04 at 10 55 09 AM

https://github.com/Expensify/App/assets/108292595/7225c0a3-20ef-4205-8d86-afcb8ef352fb

**Comment by situchan** [comment]
### Tests

#### 1. We're showing right **carets** for all **report** and **request previews**, when instead right **carets** should be removed from all **request** and report **previews**.


https://github.com/Expensify/App/assets/108292595/ee18a380-c2a4-409a-b19a-88586320ea3a


#### 2. Similarly, `IOU` **reports** are showing the **merchant** when they should instead be showing the **description** (_because `IOUs` always have the same **merchant**, which is **Request_**).

https://github.com/Expensify/App/assets/108292595/8c5ba860-0e60-4212-97ed-58e9882a6907



#### 3. When a **merchant** name exists for an **expense** / **request preview**, we only show the **merchant**.

https://github.com/Expensify/App/assets/108292595/a5075953-e8f6-4599-8b4b-3a863c3f1378



#### 4. Normal "Cash" requests are missing the `X requests` logic.

https://github.com/Expensify/App/assets/108292595/a511b2b4-f1f7-4b96-bfbe-1f6903723072



#### 5. Receipt whisper cards are no visible when they are inside "Only visible for you" message.

https://github.com/Expensify/App/assets/108292595/a88f1a3a-310e-4b84-aa30-7e84eb001c63



#### 6. (none) is showing as a merchant for receipts being scanned

https://github.com/Expensify/App/assets/108292595/54575119-ddd2-4c5c-b694-155aa77ca2ef



#### 7. Rounded bottom corners of the images in the report preview

https://github.com/Expensify/App/assets/108292595/ff46d961-19a0-4c60-a953-583bc7ded7ff



#### 8. On Report Preview for requests made to Saastr or SBE workspaces, there is no user email in the workspace name

Screenshot 2023-09-03 at 9 11 20 PM

Screenshot 2023-09-03 at 9 11 26 PM


Screenshot 2023-09-03 at 9 11 11 PM

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/pull/26603#issuecomment-1704615614

</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://github.com/Expensify/App/assets/108292595/890ac91f-d51d-4012-8a17-d6929df016c2



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://github.com/Expensify/App/assets/108292595/f2999b35-5977-4621-b51a-be50c859440b



</details>

<details>
<summary>Desktop</summary>


https://github.com/Expensify/App/assets/108292595/941cb6a7-b166-4f32-bdef-c08140aa7be2



</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/108292595/9360806f-9623-4ec7-adc9-3812a064d5dc



</details>

<details>
<summary>Android</summary>


https://github.com/Expensify/App/assets/108292595/5eb04bc0-728e-41ff-98b6-be5b16953c8f



</details>

**Comment by situchan** [comment]
Bug: [mobile] hover flickering on long-press (happening on main)


https://github.com/Expensify/App/assets/108292595/bdc47930-b747-4789-9c82-924db543b533

**Review by situchan — APPROVED** [review]
:shipit:

**Review by Julesssss — APPROVED** [review]
Tests well for me 👍

**Comment by shawnborton** [comment]
Something still feels off here... the corners of the receipt don't feel very crisp:
image


I also wanted to confirm that the inner border on those receipt images is 4px?

File: src/components/ReportActionItem/MoneyRequestPreview.js [hunk 2]
```diff
@@ -156,6 +158,11 @@ function MoneyRequestPreview(props) {
     const hasFieldErrors = TransactionUtils.hasMissingSmartscanFields(props.transaction);
     const isDistanceRequest = TransactionUtils.isDistanceRequest(props.transaction);
 
+    // Show the merchant for IOUs and expenses only if they are custom or not related to scanning smartscan
+    const shouldShowMerchant =
+        !_.isEmpty(requestMerchant) && !props.isBillSplit && requestMerchant !== CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT && requestMerchant !== CONST.TRANSACTION.DEFAULT_MERCHANT;
+    const shouldShowDescription = !_.isEmpty(description) && !shouldShowMerchant;
```
Comment (allroundexperts) [inline src/components/ReportActionItem/MoneyRequestPreview.js:164]: We should have checked if the receipt was not being scanned here. Not doing so caused https://github.com/Expensify/App/issues/31104.

Comment (rezkiy37) [reply src/components/ReportActionItem/MoneyRequestPreview.js:164]: Okay, thank you for letting me know. AFAIK, the bug already fixed.

**Pull request merged**